lib/JSON/Validator.pm

```diff
@@ -900,6 +902,22 @@ sub _validate_type_object {
     delete $rules{$k};
   }
 
+  for my $data_key (keys %{$schema->{dependencies} || {}}) {
+    next if not exists $data->{$data_key};
+    if (ref $schema->{dependencies}{$data_key} eq 'ARRAY') {
+      for my $dep_key (@{$schema->{dependencies}{$data_key} || []}) {
+        next if exists $data->{$dep_key};
+        push @errors, E json_pointer($path, $dep_key),
+          [object => dependencies => $data_key];
+      }
+    }
+    elsif (ref $schema->{dependencies}{$data_key} eq 'HASH') {
+      push @errors,
+        $self->_validate_type_object($data, $path,
+        $schema->{dependencies}{$data_key});
+    }
+  }
```

Comment:
Sorry for going back/forward on this code, but I think this is easier to read:

```perl
  my $dependencies = $schema->{dependencies} || {};
  for my $k (keys %$dependencies) {
    next if not exists $data->{$k};

    if (ref $dependencies->{$k} eq 'ARRAY') {
      push @errors,
        map { E json_pointer($path, $_), [object => dependencies => $k]; }
        grep { !exists $data->{$_} } @{$dependencies->{$k}};
    }
    elsif (ref $dependencies->{$k} eq 'HASH') {
      push @errors,
        $self->_validate_type_object($data, $path, $dependencies->{$k});
    }
  }
```

Reply:
the map and grep is not easier to read.

Reply:
..but ok.